core/src/main/java/com/graphhopper/routing/weighting/custom/CustomWeightingHelper.java

```diff
@@ -57,6 +59,14 @@ protected final double getRawSpeed(EdgeIteratorState edge, boolean reverse) {
         return speed;
     }
 
+    protected final double getRawPriority(EdgeIteratorState edge, boolean reverse) {
+        if (priority_enc == null) return 1;
+        double priority = reverse ? edge.getReverse(priority_enc) : edge.get(priority_enc);
+        if (Double.isInfinite(priority) || Double.isNaN(priority) || priority < 0)
+            throw new IllegalStateException("Invalid priority " + priority);
```

Comment:
Can we / should we also check `priority < 1.5` here? And don't we have to adjust `CustomWeighting#getMinWeight` as well, now that priority can be max. `1.5` in case we do not apply any rules in custom model?

Reply:
The priority above 1 was possible for server-side profiles in versions before (3.0) so I consider this a different issue.

Reply:
Ok, but now that CustomWeighting does use priority and now that priority can be 1.5 I think we have to adjust `CustomWeighting#getMinWeight` otherwise the routing could be wrong, don't you think?

Reply:
Have created #2334

Reply:
Ok thanks :+1:. Before this PR we did not have to take into account priority in `CustomWeighting#getMinWeight`, because CustomWeighting simply ignored the `vehicle$priority` values and priority was always `<1`, right? With this PR it should be easy to construct a test case that produces connection not found if we set `Priority.BEST`?

Reply:
> and priority was always <1, right?

Before this PR when I created a JSON that increased the priority above 1 then this was also a problem.

Reply:
> With this PR it should be easy to construct a test case that produces connection not found if we set Priority.BEST?

It will find the connection, I think, but could give a suboptimal route as the heuristic with A* is no longer underestimate

Reply:
> Before this PR when I created a JSON that increased the priority above 1 then this was also a problem.

Ok in the server-side config producing this problem was also possible, right. 

> It will find the connection, I think, but could give a suboptimal route as the heuristic with A* is no longer guaranteed to be underestimating

I think with edge-based CH AStar it is also possible to get connection not found, but I will try to come up with a test case to proof this :)

And fixing this is actually quite easy, right? We just need `encoder#getMaxPriority` like `encoder#getMaxSpeed` and need to use this in `getMinWeight()`? But sure we can also do this in a follow-up PR.

Reply:
> I think with edge-based CH AStar it is also possible to get connection not found, but I will try to come up with a test case to proof this :)

Aha, ok. No need to proof this - I just believe you :D ... I think finding those problems is rather time consuming and not sure if worth the effort :) 

> And fixing this is actually quite easy, right? We just need encoder#getMaxPriority like encoder#getMaxSpeed and need to use this in getMinWeight()? But sure we can also do this in a follow-up PR.

Yes, see https://github.com/graphhopper/graphhopper/pull/2335 :) (where we would replace the 1 with priorityEnc.getMax or something to integrate it here)

Reply:
> Aha, ok. No need to proof this - I just believe you :D ... I think finding those problems is rather time consuming and not sure if worth the effort :)

Ah I think actually quite the opposite. I probably spent too much time thinking and talking about all the possibilities and it is more efficient  (and sustainable) to start with a failing test to identify the (or one of the) problems :)

>  Yes, see #2335 :) (where we would replace the 1 with priorityEnc.getMax or something to integrate it here)

:+1: